cmake/Codegen.cmake

```diff
@@ -55,7 +55,10 @@ install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
 if (INTERN_BUILD_ATEN_OPS)
   SET(OPT_FLAG "-O3 ")
   IF(MSVC)
-    SET(OPT_FLAG "/Ox /fp:strict ")
+	IF(NOT "${CMAKE_BUILD_TYPE}" MATCHES "Debug")
```

Comment:
Well, it is still relying on `CMAKE_BUILD_TYPE`, which is empty by default for VS. A better solution is to use the generator expressions. You can refer to https://github.com/pytorch/pytorch/pull/18548.

Reply:
Do we want to support all 4 default build types including `RELWITHDEBINFO` & `MINSIZEREL`, or limit only to debug/release?

Reply:
Is there any difficulty in supporting all the types here? If no, then let's go for it.

Reply:
see 76cf0a7